exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -314,54 +311,25 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 			Applied:         false,
 			LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
 		})
-		// Since maps are not ordered, we need to order them to get the same hash at each reconcile.
-		keys := make([]string, 0)
-		data, ok := unstructuredObj.UnstructuredContent()["data"]
-		if !ok {
-			errList = append(errList, errors.New("failed to get data field from the resource"))
-			continue
-		}
-
-		unstructuredData := data.(map[string]interface{})
-		for key := range unstructuredData {
-			keys = append(keys, key)
-		}
-		sort.Strings(keys)
-
-		dataList := make([][]byte, 0)
-		for _, key := range keys {
-			val, ok, err := unstructured.NestedString(unstructuredData, key)
-			if !ok || err != nil {
-				errList = append(errList, errors.New("failed to get value field from the resource"))
-				continue
-			}
 
-			byteArr := []byte(val)
-			// If the resource is a Secret, data needs to be decoded.
-			if unstructuredObj.GetKind() == string(addonsv1.SecretClusterResourceSetResourceKind) {
-				byteArr, _ = base64.StdEncoding.DecodeString(val)
-			}
-
-			dataList = append(dataList, byteArr)
+		if err != nil {
```

Comment:
I think the reason why it's done like this is that we always want to call `resourceSetBinding.SetBinding`

Reply:
That being said. What about slightly more code redundancy in favor of better readable code, e.g.:
```go
		resourceScope, err := reconcileScopeForResource(clusterResourceSet, resource, resourceSetBinding, unstructuredObj)
		if err != nil {
			// Set status in ClusterResourceSetBinding to surface that the resource is not applied.
			resourceSetBinding.SetBinding(addonsv1.ResourceBinding{
				ResourceRef:     resource,
				Hash:            "",
				Applied:         false,
				LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
			})

			errList = append(errList, err)
			continue
		}

		if !resourceScope.needsApply() {
			// Nothing to do.
			continue
		}

		// Set status in ClusterResourceSetBinding in case of early continue due to a failure.
		// Set only when resource is retrieved successfully.
		resourceSetBinding.SetBinding(addonsv1.ResourceBinding{
			ResourceRef:     resource,
			Hash:            "",
			Applied:         false,
			LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
		})

```

Reply:
@sbueringer sounds like a good tradeoff, changed. Thanks!

Reply:
Looks good!